network/wsPeer.go

```diff
@@ -410,7 +411,7 @@ func (wp *wsPeer) Respond(ctx context.Context, reqMsg IncomingMessage, responseT
 	}
 
 	select {
-	case wp.sendBufferBulk <- sendMessages{msgs: msg}:
+	case wp.sendBufferBulk <- sendMessages{msgs: msg, callback: reqMsg.Callback}:
```

Comment:
If this select case is not picked, you may leak the counter decrement. 
For now, this likely won't matter, since ctx is canceled only when the service is shutting down.
If this is the case, we are still vulnerable to future changes. 
Better to handle the case here and make the behavior robust irrespective of why or who closes or cancels.

Reply:
Agreed, thanks!